### PR TITLE
CR-9610

### PR DESCRIPTION
**Pull request by elad-codefresh**

### CODE REVIEW

**Review by roi-codefresh — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -38,6 +38,7 @@ require (
 )
 
 replace (
+	github.com/argoproj-labs/argocd-autopilot => /Users/elad-codefresh/work/argocd-autopilot
```
Comment (elad-codefresh) [inline go.mod]: will remove

File: pkg/store/store.go [hunk 2]
```diff
@@ -219,6 +222,7 @@ func init() {
 	s.NetworkTesterImage = "quay.io/codefresh/cf-venona-network-tester:latest"
 	s.MinKubeVersion = "v1.18.0"
 	s.MaxKubeVersion = "v1.21.9"
+	s.CFInternalGitSources = []string{"marketplace-git-source"}
```
Comment (elad-codefresh) [inline pkg/store/store.go]: gs logic

File: pkg/store/store.go [hunk 3]
```diff
@@ -131,6 +132,7 @@ type Store struct {
 	NetworkTesterImage                   string
 	MinKubeVersion                       string
 	MaxKubeVersion                       string
+	CFInternalGitSources                 []string
```
Comment (elad-codefresh) [inline pkg/store/store.go:135]: gs logic

File: cmd/commands/git-source.go [hunk 4]
```diff
@@ -226,6 +226,9 @@ func RunGitSourceCreate(ctx context.Context, opts *GitSourceCreateOptions) error
 		URL:  opts.GsCloneOpts.Repo,
 	}
 
+	_, isInternal := FindIndexIfExists(store.Get().CFInternalGitSources, appDef.Name)
```
Comment (elad-codefresh) [inline cmd/commands/git-source.go]: gs logic

File: cmd/commands/git-source.go [hunk 5]
```diff
@@ -19,7 +19,7 @@ import (
 	"os"
 	"strings"
 	"time"
-
+	"strconv"
```
Comment (elad-codefresh) [inline cmd/commands/git-source.go]: gs logic

File: cmd/commands/common.go [hunk 6]
```diff
@@ -585,3 +585,13 @@ func askUserIfToProceedWithInsecure(ctx context.Context) error {
 	}
 	return nil
 }
+
+func FindIndexIfExists(slice []string, val string) (int, bool) {
```
Comment (elad-codefresh) [inline cmd/commands/common.go]: gs logic - and for future use

Comment (roi-codefresh) [reply cmd/commands/common.go]: move to pkg/util/util.go rename to `StringIndexOf`, no need to return `bool` it is a standard that those functions that return index return `-1` if not found.

File: cmd/commands/git-source.go [hunk 7]
```diff
@@ -226,6 +226,9 @@ func RunGitSourceCreate(ctx context.Context, opts *GitSourceCreateOptions) error
 		URL:  opts.GsCloneOpts.Repo,
 	}
 
+	_, isInternal := FindIndexIfExists(store.Get().CFInternalGitSources, appDef.Name)  
+	appDef.IsInternal = strconv.FormatBool(isInternal)
```
Comment (roi-codefresh) [inline cmd/commands/git-source.go]: after you fix the util function you can do:
```
appDef.IsInternal = util.StringIndexOf(store.Get().CFInternalGitSources, appDef.Name) > -1
```

File: pkg/store/store.go (hunk 2 above)
Comment (roi-codefresh) [reply pkg/store/store.go]: u can use `s.MarketplaceGitSourceName` as value here :)

**Review by roi-codefresh — APPROVED** [review]

File: cmd/commands/git-source.go [hunk 8]
```diff
@@ -444,13 +447,16 @@ func NewGitSourceListCommand() *cobra.Command {
 				return fmt.Errorf("must enter runtime name")
 			}
 
-			return RunGitSourceList(cmd.Context(), args[0])
+			return RunGitSourceList(cmd.Context(), args[0], cmd)
 		},
 	}
+
+	cmd.Flags().BoolVar(&includeInternal, "include-internal", false, "If true, will include the Codefresh-internal git-sources")
```
Comment (roi-codefresh) [inline cmd/commands/git-source.go]: very good.

Comment (roi-codefresh) [inline cmd/commands/git-source.go]:
```suggestion
	cmd.Flags().BoolVar(&includeInternal, "include-internal", false, "If true, will include the Codefresh internal git-sources")
```

File: cmd/commands/git-source.go [hunk 9]
```diff
@@ -444,13 +447,16 @@ func NewGitSourceListCommand() *cobra.Command {
...
+
 	return cmd
 }
 
-func RunGitSourceList(ctx context.Context, runtimeName string) error {
+func RunGitSourceList(ctx context.Context, runtimeName string, cmd *cobra.Command) error {
```
Comment (roi-codefresh) [inline cmd/commands/git-source.go]: why not just pass in the `bool`? I am super confused by this :/

File: cmd/commands/git-source.go [hunk 10]
```diff
@@ -474,6 +480,11 @@ func RunGitSourceList(ctx context.Context, runtimeName string) error {
 
 	for _, gs := range gitSources {
 		name := gs.Metadata.Name
+		nameWithoutRuntimePrefix := strings.Replace(name, fmt.Sprintf("%s-", runtimeName), "", 1)
+		if util.StringIndexOf(store.Get().CFInternalGitSources, nameWithoutRuntimePrefix) > -1 && !cmd.Flags().Changed("include-internal") {
```
Comment (roi-codefresh) [inline cmd/commands/git-source.go]: maybe just have a util function` isInternalGitSource(gs *GitSource..) bool`

File: cmd/commands/runtime.go [hunk 11]
```diff
@@ -868,7 +877,8 @@ func installComponents(ctx context.Context, opts *RuntimeInstallOptions, rt *run
 				version:      "v1alpha1",
 			},
 		},
-		saName: store.Get().RolloutReporterServiceAccount,
+		saName:     store.Get().RolloutReporterServiceAccount,
+		IsInternal: util.StringIndexOf(store.Get().CFInternalReporters, store.Get().RolloutReporterName) > -1,
```
Comment (roi-codefresh) [inline cmd/commands/runtime.go]: I don't understand this. it worked without it, why do we need this?

Comment (elad-codefresh) [reply cmd/commands/runtime.go]: I don't think it worked without it, it was one of the things I added, in order to include the label in the createReporter func

File: cmd/commands/git-source.go (hunk 10 above)
Comment (elad-codefresh) [reply cmd/commands/git-source.go]: won't this (StringIndexOf) be a useful function for possible future uses?

File: cmd/commands/git-source.go (hunk 9 above)
Comment (elad-codefresh) [reply cmd/commands/git-source.go]: I can pass the bool instead, and do this operation one level sooner, np

File: cmd/commands/git-source.go [hunk 12]
```diff
@@ -474,6 +480,11 @@ func RunGitSourceList(ctx context.Context, runtimeName string) error {
 
 	for _, gs := range gitSources {
 		name := gs.Metadata.Name
+		nameWithoutRuntimePrefix := strings.Replace(name, fmt.Sprintf("%s-", runtimeName), "", 1)
```
Comment (roi-codefresh) [inline cmd/commands/git-source.go]: strings.StripPrefix is a better fit

File: cmd/commands/runtime.go (hunk 11 above)
Comment (roi-codefresh) [reply cmd/commands/runtime.go]: what I don't understand is - which reporter is not internal? I thin this should be assumed that all of them are and just set it to true inside `createReporter` func

File: cmd/commands/git-source.go (hunk 12 above)
Comment (elad-codefresh) [reply cmd/commands/git-source.go]: I see that StripPrefix is part of net/http and not strings lib, am I missing something?

File: cmd/commands/runtime.go [hunk 13]
```diff
@@ -1651,9 +1662,10 @@ func createEventsReporter(ctx context.Context, cloneOpts *git.CloneOptions, opts
 
 	resPath := cloneOpts.FS.Join(apstore.Default.AppsDir, store.Get().EventsReporterName, opts.RuntimeName, "resources")
 	appDef := &runtime.AppDef{
-		Name: store.Get().EventsReporterName,
-		Type: application.AppTypeDirectory,
-		URL:  cloneOpts.URL() + "/" + resPath,
+		Name:       store.Get().EventsReporterName,
+		Type:       application.AppTypeDirectory,
+		URL:        cloneOpts.URL() + "/" + resPath,
+		IsInternal: util.StringIndexOf(store.Get().CFInternalReporters, store.Get().EventsReporterName) > -1,
```
Comment (roi-codefresh) [inline cmd/commands/runtime.go]: just `true` here.

**Review by roi-codefresh — APPROVED** [review]

**Pull request merged**